src/components/transactions/TxFilterForm/utils.ts

```diff
@@ -3,60 +3,128 @@ import {
   TxFilterType,
   type IncomingTxFilter,
   type ModuleTxFilter,
-  type OutgoingTxFilter,
-  type TxFilter,
+  type MultisigTxFilter,
   type TxFilterFormState,
+  type TxFilterQuery,
 } from '@/components/transactions/TxFilterForm/types'
+import { isDate, isString, omit } from 'lodash'
+import { ParsedUrlQuery } from 'querystring'
+import { TxFilterFormFieldNames } from '.'
+import { getIncomingTransfers, getMultisigTransactions, getModuleTransactions } from '@gnosis.pm/safe-react-gateway-sdk'
+
+const TX_FILTER_FIELD_NAMES = Object.values(TxFilterFormFieldNames)
+const TX_FILTER_TYPES = Object.values(TxFilterType)
+
+export const _hasTxFilterType = <T extends ParsedUrlQuery | TxFilterFormState>(
+  query: T,
+): query is T & { [TxFilterFormFieldNames.FILTER_TYPE_FIELD_NAME]: TxFilterType } => {
+  const type = query[TxFilterFormFieldNames.FILTER_TYPE_FIELD_NAME]
+  return isString(type) && TX_FILTER_TYPES.includes(type as TxFilterType)
+}
+
+export const hasTxFilterQuery = (query: ParsedUrlQuery): boolean => {
+  return (
+    _hasTxFilterType(query) &&
+    Object.keys(query).some((key) => isString(key) && TX_FILTER_FIELD_NAMES.includes(key as TxFilterFormFieldNames))
+  )
+}
+
+export const getFilterlessQuery = (query: ParsedUrlQuery): ParsedUrlQuery => {
```

Comment:
Please call it `omitTxFilterQuery` or something. Filterless doesn't sound computery enough.